generator/osm_element.hpp

```diff
@@ -109,7 +109,7 @@ struct OsmElement
   bool HasTag(std::string const & key, std::string const & value) const;
 
   template <class Fn>
-  void UpdateTag(std::string const & key, Fn && fn)
+  void UpdateTagFn(std::string const & key, Fn && fn)
```

Comment:
I wanted to overload UpdateTag() to work both with lambdas and strings, but didn't succeed (for some reason the compiler ignored the explicit string version and continued to use the lambda version for string arguments which failed of course).
Would be glad if someone could do it properly.

Reply:
How did you overload it and use it?

Reply:
```
  template <class Fn>
  void UpdateTag(std::string const & key, Fn && fn)
  {
   .... use lambda like fn(value);
  }
  void UpdateTag(std::string const & key, std::string const & value)
  {
    UpdateTag(key, [&value](auto & v) { v = value; });
  }
```
gives errors like
```
/home/pastk/dev/OM/src/generator/osm_element.hpp: In instantiation of ‘void OsmElement::UpdateTag(const std::string&, Fn&&) [with Fn = std::__cxx11::basic_string<char>&; std::string = std::__cxx11::basic_string<char>]’:
/home/pastk/dev/OM/src/generator/osm2type.cpp:863:21:   required from here
/home/pastk/dev/OM/src/generator/osm_element.hpp:118:11: error: no match for call to ‘(std::__cxx11::basic_string<char>) (std::string&)’
  118 |         fn(tag.m_value);
      |         ~~^~~~~~~~~~~~~
/home/pastk/dev/OM/src/generator/osm_element.hpp:124:7: error: no match for call to ‘(std::__cxx11::basic_string<char>) (std::string&)’
  124 |     fn(value);
      |     ~~^~~~~~~
/home/pastk/dev/OM/src/generator/osm_element.hpp: In instantiation of ‘void OsmElement::UpdateTag(const std::string&, Fn&&) [with Fn = const char (&)[2]; std::string = std::__cxx11::basic_string<char>]’:
/home/pastk/dev/OM/src/generator/osm2type.cpp:946:17:   required from here
/home/pastk/dev/OM/src/generator/osm_element.hpp:118:11: error: expression cannot be used as a function
  118 |         fn(tag.m_value);
      |         ~~^~~~~~~~~~~~~
/home/pastk/dev/OM/src/generator/osm_element.hpp:124:7: error: expression cannot be used as a function
  124 |     fn(value);
      |     ~~^~~~~~~
```
I've tried some other options as well like
```
  template < >
  void UpdateTag<std::string>(std::string const & key, std::string const & value)
```

It'd be better if someone knowledgeable just does it and sends in a working code :)